app/src/verifier.h

```diff
@@ -362,17 +363,106 @@ namespace scitt::verifier
         }
         else if (phdr.issuer.has_value())
         {
-          // IETF SCITT claim
-          key = process_ietf_profile(
-            phdr, tx, current_time, resolution_cache_expiry, configuration);
-
-          try
+          if (phdr.issuer->starts_with("did:x509"))
           {
-            cose::verify(data, key);
+            // IETF SCITT did:x509 claim
+            if (phdr.x5chain.has_value())
+            {
+              // NB: In later revisions of SCITT, x5chain is unprotected, and
+              // only x5t is. This logic will need to authenticate x5chain[0]
+              // against x5t before it can proceed to verify the signature.
+              // Verify the signature as early as possible
+              OpenSSL::Unique_X509 leaf =
+                parse_certificate(phdr.x5chain.value()[0]);
+              key = PublicKey(leaf, std::nullopt);
+              try
+              {
+                cose::verify(data, key);
+              }
+              catch (const cose::COSESignatureValidationError& e)
+              {
+                throw VerificationError(e.what());
+              }
+              // Then authenticate the did:x509 claim against the x5chain
+              std::string pem_chain;
+              for (auto const& c : phdr.x5chain.value())
+              {
+                pem_chain += ccf::crypto::cert_der_to_pem(c).str();
+              }
+              auto did_document_str = didx509::resolve(
+                pem_chain,
+                phdr.issuer.value(),
+                true /* Do not validate time */);
+              scitt::did::alt::DIDDocument did_document =
+                nlohmann::json::parse(did_document_str);
+
+              if (did_document.verification_method.empty())
```

Comment:
Why only verification method is checked. In the did:web case we use only assetion method. I would expect both to be used here for verififcation `assertionMethod` and `verificationMethod`

Reply:
Because did:x509 does not have assertionMethod? I am not sure what we would check there, can you clarify?

Reply:
As discussed, this is not necessary because there is only a ever a single key. We can simplify the interface for optimisation purposes by making a change in the library and propagating through CCF to here (logged as https://github.com/microsoft/didx509cpp/issues/16).